### PR TITLE
[Rancher v2.6.5] Bump the default rke, rke2, and k3s to 1.23 

### PR DESCRIPTION
**Pull request by jiaqiluo**
Issue: https://github.com/rancher/rancher/issues/37432

This PR is to bump the default rke, rke2, and k3s to 1.23 for Rancher v2.6.5,
it also includes a small fix for ignoring the annoying DS_Store file in macOS.

### CODE REVIEW

File: channels-rke2.yaml
Comment (jakefhyde) [inline channels-rke2.yaml]: I feel like one of these should be an inclusive range. I know we never ship with a `-0` (and `< 2.6.5-0` is effectively synonymous with `<= 2.6.4` since the `-0` prerelease identifier in `2.6.5-0` means "the earliest possible thing that could be considered part of the `2.6.5` release line), but according to strict semver, this means that `2.6.5-0` is excluded (i.e. has no default), and 2.6.5-rc1 is included for 1.23.x. While we will never ship a `-0` version (IIRC the earliest thing used is always an `-alpha1`, we also do some weird things in rancher internally regarding prerelease versions. If 1.23.x should be the default for `v2.6.5-rc1` and up, I think it should be `appVersion: '>= 2.6.5-0 < 2.6.99-0'` instead so it's explicit which boundary it should be including.

Comment (jiaqiluo) [reply channels-rke2.yaml]: Good points! PR is updated, please review it again.

**Review by jakefhyde — DISMISSED** [review]

File: rke/k8s_version_info.go
Comment (snasovich) [inline rke/k8s_version_info.go]: Is this safe to merge this change into `dev-v2.6` branch provided we release `release-v2.6` from it? The other changes should be fine I think, just concerned about this one.
@kinarashah

Comment (kinarashah) [reply rke/k8s_version_info.go]: @snasovich Since this issue is for v2.6.5 milestone, I was under the impression that we would hold off on merging this PR till April patches are released. I don't think we want to mark v1.23 as default here until the local cluster is updated in rancher's codebase with v1.23 (which would be released with v2.6.5)

Comment (jiaqiluo) [reply rke/k8s_version_info.go]: @kinarashah @snasovich 
The April patch was released. So this PR can be reviewed again. 
There is [another PR](https://github.com/rancher/rancher/pull/37453) for updating the local cluster k8s version in rancher, and I will make sure that PR is merged first.

Comment (snasovich) [inline rke/k8s_version_info.go]: We now have `v1.23.6-rancher1-1` following the merge of April 2022 k8s patches, so I think it should be updated here.

**Review by snasovich — CHANGES_REQUESTED** [review]
One minor comment that I believe needs to be addressed before we're good to merge.

File: rke/k8s_version_info.go
Comment (jiaqiluo) [reply rke/k8s_version_info.go]: fixed.

**Review by snasovich — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**
